xgi/communities/spectral.py

```diff
@@ -0,0 +1,126 @@
+import numpy as np
+from scipy.sparse.linalg import eigsh
+
+from ..exception import XGIError
+from ..linalg.laplacian_matrix import normalized_hypergraph_laplacian
+
+__all__ = [
+    "spectral_clustering",
+]
+
+
+def spectral_clustering(H, k=2, **kwargs):
+    """Cluster hypergraph into k-many groups of nodes using spectral techniques.
+
+    Compute a spectral clustering according to the heuristic suggested in [1].
```

Comment:
I would suggest to make this more explanatory, see below for a suggestion (please check that everything is correct and feel free to change it however you think it's better)

```suggestion
    """Computes a spectral clustering in :math:`k` partitions of the input 
    hypergraph according to the heuristic presented in [1].
    
    This is done by computing the normalized Laplacian of the input hypergraph 
    and then applying :math:`k`-means clustering on the hypergraph vertices 
    represented in the :math:`k`-dimensional Euclidian space generated 
    by the eigenvectors corresponding to the :math:`k` largest eigenvalues of the 
    normalized Laplacian.
     
 ```

Reply:
This is mostly the procedure, albeit with the smallest eigenvectors not the largest. I am adding it to the PR now